doc/how-to/app/app_roles.rst

```diff
@@ -0,0 +1,476 @@
+
+
+
+
+.. _roles_create_custom_role_apply:
+
+Applying a role configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To apply the validated configuration, define the :ref:`apply([cfg]) <roles_api_reference_apply>` function.
+As the ``validate()`` function, ``apply()`` provides access to a role's configuration using the ``cfg`` argument.
+
+In the example below, the ``apply()`` function uses the :ref:`log <log-module>` module to write a role's configuration value to the log:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/application_role_cfg/greeter.lua
+    :language: lua
+    :start-at: local function apply
+    :end-before: local function stop
+    :dedent:
+
+
+
+.. _roles_create_custom_role_stop:
+
+Stopping a role
+~~~~~~~~~~~~~~~
+
+To stop a role, use the :ref:`stop() <roles_api_reference_stop>` function.
+
+In the example below, the ``stop()`` function uses the :ref:`log <log-module>` module to indicate that a role is stopped:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/application_role_cfg/greeter.lua
+    :language: lua
+    :start-at: local function stop
+    :end-before: return
+    :dedent:
+
+When you've defined all the role's functions, you need to return an object that has corresponding functions specified:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/application_role_cfg/greeter.lua
+    :language: lua
+    :start-at: return
+    :dedent:
+
+
+
+.. _roles_create_custom_role_dependencies:
+
+Role dependencies
+~~~~~~~~~~~~~~~~~
+
+To define a role's dependencies, use the :ref:`dependencies <roles_api_reference_dependencies>` field.
+In this example, the ``byeer`` role has the ``greeter`` role as the dependency:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/application_role_cfg/byeer.lua
+    :language: lua
+    :dedent:
+
+A role cannot be started without its dependencies.
+This means that all the dependencies of a role should be defined in the ``roles`` configuration parameter:
+
+..  code-block:: yaml
+
+    instance001:
+      roles: [ greeter, byeer ]
+
+You can find the full example here: `application_role_cfg <https://github.com/tarantool/doc/tree/latest/doc/code_snippets/snippets/config/instances.enabled/application_role_cfg>`_.
+
+
+
+.. _roles_create_custom_role_init:
+
+Adding initialization code
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can add initialization code to a role by defining and calling a function with an arbitrary name at the top level of a module, for example:
+
+..  code-block:: lua
+
+    local function init()
+        -- ... --
+    end
+
+    init()
+
+For example, you can :ref:`create spaces <box_schema-space_create>`, define :ref:`indexes <concepts-data_model_indexes>`, or :ref:`grant privileges <configuration_credentials>` to specific users or roles.
+
+See also: :ref:`roles_create_space`.
+
+
+
+.. _roles_create_space:
+
+Specifics of creating spaces
```

Comment:
```suggestion
Creating spaces in roles
```

Reply:
This applies to creating spaces for applications in general. So, I'm going to move this section outside this topic at some moment.